src/context/QuestsContext.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react'
+import { useState } from 'react'
+
+const QuestContext = React.createContext(null)
+
+const DEFAULT_STATE = [
+	{
+		id: 1,
+		name: 'quest1',
+		isComplete: false,
+		isUnlocked: true,
+	},
+	{ id: 2, name: 'quest2', isComplete: false, isUnlocked: false },
+]
+
+const QuestProvider = ({ children }) => {
+	const getQuests = () => {
+		const questString = localStorage.getItem('quests')
+		const questMod = questString ? JSON.parse(questString) : DEFAULT_STATE
+		localStorage.setItem('quests', JSON.stringify(questMod))
+
+		return questMod
+	}
+
+	const getQuestCounter = () => {
+		const questString = localStorage.getItem('quest_number')
+		const questMod = questString ? JSON.parse(questString) : 1
+
+		localStorage.setItem('quest_number', JSON.stringify(questMod))
+
+		return questMod
+	}
+
+	const [quests, setQuests] = useState(getQuests())
+	const [questCounter, updateQuestCounter] = useState(getQuestCounter())

```

Comment:
The `questCounter` naming is misleading and in general - I'm not convinced why overcomplicating it with storing this value is needed.